politeiad/backendv2/tstorebe/plugins/comments/comments.go

```diff
@@ -102,12 +104,199 @@ func (p *commentsPlugin) Hook(h plugins.HookT, payload string) error {
 func (p *commentsPlugin) Fsck(tokens [][]byte) error {
 	log.Tracef("comments Fsck")
 
-	// Verify the cohereny of the record index for each token. This
-	// includes verifying that:
-	// - All comment adds are present in the record index.
-	// - All comment dels are present in the record index and the
-	//   corresponding comment adds have been deleted from the db.
-	// - All comment votes are present in the record index.
+	log.Infof("Starting comments fsck for %v records", len(tokens))
+
+	counter := 0 // number of rebuilded record indexes
+
+	// Navigate the provided record tokens and verify their record index
+	// comments cache integrity. This will only rebuild the cache if any
+	// inconsistency is found.
+	for _, token := range tokens {
```

Comment:
Let's set this up a little differently in order to improve readability.

```
	for _, token := range tokens {
		// Verify the coherency of the record index.
		isCoherent, err := p.verifyRecordIndex(token)
		if err != nil {
			return err
		}
		if isCoherent {
			continue
		}

		// The record index is not coherent and needs to be rebuilt.
		err = p.rebuildRecordIndex(token)
		if err != nil {
			return err
		}
	}
```

I would put the `p.verifyRecordIndex` and `p.rebuildRecordIndex` functions in a `fsck.go` file in order to prevent this main `comments.go` file from becoming cluttered.

Reply:
awesome, it'll be much better like that. on it